samples/openvx/openvx_interoperability.cpp

```diff
@@ -0,0 +1,324 @@
+    vx_status status;
+    vx_image ovxImage = vxCreateImage(context, width, height, format);
+    status = vxGetStatus((vx_reference)ovxImage);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to create image");
+    return ovxImage;
+}
+
+vx_image createVirtualImage(vx_graph graph, vx_uint32 width, vx_uint32 height, vx_df_image format)
+{
+    vx_status status;
+    vx_image ovxImage = vxCreateVirtualImage(graph, width, height, format);
+    status = vxGetStatus((vx_reference)ovxImage);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to create virtual image");
+    return ovxImage;
+}
+
+void getValidRegionImage(vx_image image, vx_rectangle_t *rect)
+{
+    vx_status status;
+    status = vxGetValidRegionImage(image, rect);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to get valid region");
+}
+
+void accessImagePatch(vx_image image,
+                        const vx_rectangle_t *rect,
+                        vx_uint32 plane_index,
+                        vx_imagepatch_addressing_t *addr,
+                        void **ptr,
+                        vx_enum usage)
+{
+    vx_status status;
+    status = vxAccessImagePatch(image, rect, plane_index, addr, ptr, usage);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to access image patch");
+}
+
+void commitImagePatch(vx_image image,
+                      vx_rectangle_t *rect,
+                      vx_uint32 plane_index,
+                      vx_imagepatch_addressing_t *addr,
+                      const void *ptr)
+{
+    vx_status status;
+    status = vxCommitImagePatch(image, rect, plane_index, addr, ptr);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to commit image patch");
+}
+
+void queryImage(vx_image image, vx_enum attribute, void *ptr, vx_size size)
+{
+    vx_status status;
+    status = vxQueryImage(image, attribute, ptr, size);
+    if(status != VX_SUCCESS) throw std::runtime_error("Failed to query image");
+}
+
+}
+
+
+vx_image convertCvMatToVxImage(vx_context context, cv::Mat image)
+{
+    if(!(!image.empty() && image.dims <= 2 && image.channels() == 1))
+        throw std::runtime_error("Invalid format");
+
+    vx_uint32 width  = image.cols;
+    vx_uint32 height = image.rows;
+
+    vx_df_image color;
+    switch (image.depth())
+    {
+    case CV_8U:
+        color = VX_DF_IMAGE_U8;
+        break;
+    case CV_16U:
+        color = VX_DF_IMAGE_U16;
+        break;
+    case CV_16S:
+        color = VX_DF_IMAGE_S16;
+        break;
+    case CV_32S:
+        color = VX_DF_IMAGE_S32;
+        break;
+    default:
+        throw std::runtime_error("Invalid format");
+        break;
+    }
+
+    vx_image ovxImage = ovx::createImage(context, width, height, color);
+
+    vx_rectangle_t rect;
+    ovx::getValidRegionImage(ovxImage, &rect);
+
+    vx_imagepatch_addressing_t addr;
+    vx_uint8* ovxData = NULL;
+    ovx::accessImagePatch(ovxImage, &rect, 0, &addr, (void**)&ovxData, VX_WRITE_ONLY);
```

Comment:
This `cv::Mat -> vx_image` copy might be done in one call with `vxCopyImagePatch` method.